site-src/geps/gep-724.md

```diff
@@ -0,0 +1,426 @@
+  // only kind of target resource with "Core" support is Gateway. This API may
+  // be extended in the future to support additional kinds of parent resources
+  // such as Routes.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=16
+  AttachTo []AttachRef `json:"gateways,omitempty"`
+```
+
+And the struct that references:
+```go
+// AttachRef identifies an API object to attach to. The only kind of target
+// resource with "Core" support is Gateway. This API may be extended in the
+// future to support additional kinds of parent resources such as Routes.
+type AttachRef struct {
+  // Group is the group of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=networking.x-k8s.io
+  Group string `json:"group"`
+
+  // Kind is kind of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Kind string `json:"kind"`
+
+  // Name is the name of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Name string `json:"name"`
+
+  // SectionName is the name of a section within the target resource. When
+  // unspecified, this targets the entire resource. In the following
+  // resources, SectionName is interpreted as the following:
+  // * Gateway: Listener Name
+  // * Route: Rule Name
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  SectionName string `json:"sectionName,omitempty"`
+
+  // Namespace is the namespace of the referent. When unspecified, the local
+  // namespace is inferred unless targeting a cluster-scoped resource in which
+  // case no namespace is inferred.
+  //
+  // Support: Extended
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  Namespace string `json:"namespace,omitempty"`
+}
+```
+
+### Advantages
+
+* Simplifies the API by providing a single way to attach Routes to Gateway.
+* Assigns clear responsibilities to Gateway and Route owners.
+* Consistent with pattern of direct references from Routes to all associated
+  resources.
+* Enables attaching Routes to arbitrary parents, such as custom Gateways, other
+  Routes (to be defined), or Meshes.
+* Prevents accidental exposure of Routes.
+* Easy to understand which Gateways/parents a Route is attached to.
+* Further simplifies path to Route inclusion.
+* Follows pattern of direct reference in one direction with a broader trust
+  reference in the other direction.
+* Aligns with initial user feedback.
+
+### Disadvantages
+
+* Attaching a Route to a named listener with SectionName may be a bit confusing.
+* Does not utilize existing ReferencePolicy mechanism.
+* May be more difficult to understand which Routes are attached to a Gateway.
+* Adding/replacing a Gateway requires changes to Routes.
+
+### Potential Expansion
```

Comment:
How does this impact possible `route -> route`?

Reply:
I think we'd end up with 3 options here, I'm not sure which is best:

1. 1:many Route->Routes: Nearly identical to Route->Gateway pattern described here, parent Route describes trusted namespaces per rule, child Routes use attachTo mechanism to attach to a parent.
2. 1:1 Route->Route: Require direct reference in both directions (include + attachTo would both be direct refs).
3. 1:1 Route->Route: Direct reference from parent to child, ReferencePolicy to denote trust from child ns.

I think 1 and 3 make the most sense, it really just depends on how we want the feature to be used. If we want Routes to be able to somewhat blindly delegate a path to a namespace, 1 makes more sense. If we want Routes to include a very specific Route in another namespace, 3 probably makes more sense.

Reply:
(3) seems to make most sense to me.

Reply:
We designed Contour's HTTPProxy to be more like (3), but we've *definitely* had requests for not requiring a specific Route name, which I think is more like (1).

I think that (3) requires more out-of-band coordination between parties than (1) does, which could be good or bad. I'm still inclined to err on the side of more coordination and explicit config, because blanked includes scare me a bit. But it's not a strong preference.

Reply:
Yeah the more I've thought about this, the more I think 3 is the best approach. That means Route->Route would work nearly identically to Route->Service, which seems reasonable to me.